hivemind/optim/optimizer.py

```diff
@@ -241,6 +244,18 @@ def __init__(
         self.tracker = self._make_progress_tracker(
             target_batch_size, performance_ema_alpha=performance_ema_alpha, **tracker_opts or {}
         )
+        if grad_rank_averager == "power_ef" and not use_local_updates:
+            assert len(extra_tensors) == 0
+            grad_extra_tensors = [
+                torch.zeros_like(param, device="cpu")
+                for param_group in optimizer.param_groups
+                for param in param_group["params"]
+            ]
+            for tensor in grad_extra_tensors:
+                if tensor is not None:
+                    tensor.share_memory_()
+            grad_averager_opts["grad_extra_tensors"] = grad_extra_tensors
+            extra_tensors = [e for e in extra_tensors] + [eg for eg in grad_extra_tensors]
```

Comment:
Let's add some boolean property of GradAverager that determines if we need to add_grads_to_state -- so that it is False for default averager but True for PowerSGD